### PR TITLE
Creeper item buff

### PR DESCRIPTION
**Pull request by bigcheesse**
Buff the creeper store item by:
* reducing cost to 30 points
* adding 2 second delay and invincibility timer
* adding CreeperRadius, CreeperTime, and CreeperShield settings
* fixing creeper blast zone position to be centered at the block the player's head is occupying.

### CODE REVIEW

File: src/main/java/org/opencraft/server/cmd/impl/CreeperCommand.java [hunk 1]
```diff
@@ -57,21 +60,27 @@ public static CreeperCommand getCommand() {
...
+        w.broadcast("- " + player.parseName() + " &eisn't feeling so good...");
+        w.broadcast("- &esssssssSSSSSSSSS");
 
-    World.getWorld().broadcast("- " + player.parseName() + " &eisn't feeling so good...");
-    World.getWorld().broadcast("- &esssssssSSSSSSSSS");
-    ((CTFGameMode)World.getWorld()
-        .getGameMode())
-        .explodeTNT(player, World.getWorld().getLevel(), px, py, pz, 4, true, true, false, null);
+        try {
+          Thread.sleep((long)(1000 * GameSettings.getFloat("CreeperTime")));
```
Comment (dakriy) [inline src/main/java/org/opencraft/server/cmd/impl/CreeperCommand.java]: If `CreeperTime < 0` Unhanded IllegalArgumentException gets thrown Maybe revert to old behavior and don't start thread if it's <= 0?

File: src/main/java/org/opencraft/server/cmd/impl/CreeperCommand.java [hunk 2]
```diff
@@ -57,21 +60,27 @@ public static CreeperCommand getCommand() {
...
+        } catch (InterruptedException ex) {}
+
+        int radius = GameSettings.getInt("CreeperRadius");
+	Position pos = player.getPosition().toBlockPos();
```
Comment (dakriy) [inline src/main/java/org/opencraft/server/cmd/impl/CreeperCommand.java]: Tabs instead of spces here

**Review by dakriy — CHANGES_REQUESTED** [review]

File: src/main/java/org/opencraft/server/cmd/impl/CreeperCommand.java (hunk 1 above)
Comment (bigcheesse) [reply src/main/java/org/opencraft/server/cmd/impl/CreeperCommand.java]: Changed so now it uses 0 if it's negative. Same for CreeperRadius.

File: src/main/java/org/opencraft/server/task/impl/CreeperTask.java
Comment (dakriy) [inline src/main/java/org/opencraft/server/task/impl/CreeperTask.java:1]: Interesting idea to setup a scheduled task for the creeper command. I think just spinning up a separate thread is probably a better approach. 2 issues with this approach as is:

1. If you do a /cr, and then a newgame or the game ends, the creeper still goes off.
2. If you kill somebody who has run /cr before it goes off, the creeper task never exits and executes infinitely until the server restarts, causing resource leaks.

File: src/main/java/org/opencraft/server/task/impl/CreeperTask.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    Position pos = this.invoker.getPosition().toBlockPos();
+
+    ctf.explodeTNT(
+      this.invoker, this.level, pos.getX(), pos.getY(), pos.getZ(),
+      radius, LETHAL, TEAMKILL, false, NAME
+    );
+
+    this.stop();
```
Comment (dakriy) [inline src/main/java/org/opencraft/server/task/impl/CreeperTask.java]: Becuse ScheduledTask is for tasks that are meant to be repeated throughout the lifetime of the server, and this only calls stop on doCreeper, If you kill somebody who has run /cr before it goes off, the creeper task never exits and executes infinitely until the server restarts, causing resource leaks.

File: src/main/java/org/opencraft/server/task/impl/CreeperTask.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    float creeperTime = GameSettings.getFloat("CreeperTime");
+    if (creeperTime > 0.0f) {
+      this.setDelay((long)(1000 * creeperTime));
+    }
+  }
+
+  public void execute() {
+    // Do not explode if the player died.
+    if (pastDeaths == invoker.deaths) {
```
Comment (dakriy) [inline src/main/java/org/opencraft/server/task/impl/CreeperTask.java]: Ignores when the game ends or changes

**Review by dakriy — CHANGES_REQUESTED** [review]
I'm liking the direction as well as all the creeper options you added. However, there are still a couple bugs that need squished.

File: src/main/java/org/opencraft/server/task/impl/CreeperTask.java (hunk 3 above)
Comment (bigcheesse) [reply src/main/java/org/opencraft/server/task/impl/CreeperTask.java]: My bad, the task should always be stopped regardless of whether the explosion happens or not.

File: src/main/java/org/opencraft/server/task/impl/CreeperTask.java (hunk 4 above)
Comment (bigcheesse) [reply src/main/java/org/opencraft/server/task/impl/CreeperTask.java]: I added a flag to the player where the creeper explosion will only happen if it is true and the player hasn't disconnected.
The flag is reset when the player dies or joins a team, or when the game ends or a new one starts.

**Review by dakriy — APPROVED** [review]
Ok LGTM

**Pull request merged**